### ISSUE TITLE
Provide a appropriate value for title attribute when ResourceNotFoundException is thrown

### ISSUE THREAD

**Issue by NageswaraRaoMaridu** [crnk-project/crnk-framework#731]
The current implementation assigns the given message to both `title` and `details` attributes of `ErrorData` which is built inside the `ResourceNotFoundException`. Assigning message to title attribute doesn't makes sense and it is more suitable to details attribute.

Current response looks something like this
```
{
    "errors": [
        {
            "status": "404",
            "title": "The requested object could not be found.",
            "detail": "The requested object could not be found."
        }
    ]
}
```

It will be good if we have something like below
```
{
    "errors": [
        {
            "status": "404",
            "title": "Not Found.",
            "detail": "The requested object could not be found."
        }
    ]
}
```

**Comment by remmeier** [comment]
makes sense, we can update the constructor of ResourceNotFoundException accordingly. Potentially it should also contain a `code` as specified in https://jsonapi.org/format/#errors.

**Comment by NageswaraRaoMaridu** [comment]
@remmeier Yes, that makes sense. We can have a telescoping constructor in `ResourceNotFoundException` for supporting title and code attributes of ErrorData

**Comment by NageswaraRaoMaridu** [comment]
@remmeier I see that `CrnkExceptionMapper`  can be used in the clients to convert the error response to respective exception. In that case, it's not enough to create those exceptions with only a message (which we are doing in CrnkExceptionMapper#fromErrorResponse). Even though the server sent error response with code, title and details, the client can't get them through CrnkMappable exceptions due to this limitation. What's your thought on this?

Would you please assign this ticket to me. I started working on this.

**Comment by remmeier** [comment]
`CrnkMappableException` is able to hold a `ErrorData`. Maybe its subtype like ResourceNotFoundException should have a constructor to allow holding the entire information sent from the server to avoid losing any information.

**Comment by NageswaraRaoMaridu** [comment]
@remmeier Correct. That's what needed in all the CrnkMappable exceptions.
